sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -235,3 +238,39 @@ private[parquet] object ParquetFilters {
     }
   }
 }
+
+/**
+ * Note that, this is a hacky workaround to allow dots in column names. Currently, column APIs
+ * in Parquet's `FilterApi` only allows dot-separated names so here we resemble those columns
+ * but only allow single column path that allows dots in the names as we don't currently push
+ * down filters with nested fields.
```

Comment:
We need to explain the functions in this object `ParquetColumns` are based on the codes in `org.apache.parquet.filter2.predicate`. Thus, when upgrading the Parquet versions, we need to check whether they are still the same.